### PR TITLE
Fix/sse reconnection

### PR DESCRIPTION
**Pull request by GagahGK**
This fixes the error mentioned it #84 , where it fails check the client existence properly due to the Fetch process thrown an exception (NetworkError) when the Internet Connection is lost for a while

with this fix, it will poll the Fetch Request until the Internet Connection is available

### CODE REVIEW

File: packages/room/channel/channel.js [hunk 1]
```diff
@@ -120,28 +121,58 @@ export const createChannel = ({ api, event, peer, streams }) => {
...
-          return
-        }
-
-        // Reconnect
-        if (errorTime - this._startTime < 1000) {
+      const onError = async () => {
+        if (!navigator.onLine) {
           setTimeout(() => {
-            this._reconnect()
+            onError()
```
Comment (faiq-naufal) [inline packages/room/channel/channel.js:127]: Will `onError()` keep trying to call itself when the user is offline?

**Review by faiq-naufal — CHANGES_REQUESTED** [review]
Only one small request change.

File: packages/room/channel/channel.js (hunk 1 above)
Comment (GagahGK) [reply packages/room/channel/channel.js:127]: it will keep trying call onError() while the user is still offline in every one second, once the connection is gained it will check the client existence and stops the polling

should we add a 5 seconds timeout to it, if it's has been out of connection for that given time it will automatically emit the `Channel_Closed` event ? with `TIMEOUT` as the reason ? since the Hub api automatically kick clients that has not been connected to the SSE for more than 5 seconds

Comment (faiq-naufal) [reply packages/room/channel/channel.js:127]: I think 1 second is fine because onError only process the part of checking the `navigator.onLine` before it is recalled again. When the user is online it will not be called again until channel error again.

Comment (faiq-naufal) [reply packages/room/channel/channel.js:127]: When the client is kicked, the reason will be `NOT_FOUND`. 

Or should we just change `NOT_FOUND` into `TIMEOUT` instead? I think `TIMEOUT` reason is better naming because when the client is kicked, the connection will be lost won't it?  And, when the connection is lost, there will be timeout until 5 seconds. The connection will be `TIMEOUT` and should be closed.

Comment (GagahGK) [reply packages/room/channel/channel.js:127]: ah what i mean about the 5 second timeout is not to change the polling rate from 1 second to 5, but add additional checks, to emit the Channel_Closed - TIMEOUT if the code has been polling the `onError()` function for more than 5 second

Comment (faiq-naufal) [reply packages/room/channel/channel.js:127]: I see, that's fine for me

**Comment by GagahGK** [comment]
@faiq-naufal I've updated the code with the timeout event if the eventSource tries to reconnect after 5 second

File: packages/room/channel/channel.js [hunk 2]
```diff
@@ -122,26 +123,55 @@ export const createChannel = ({ api, event, peer, streams }) => {
...
           this._event.emit(RoomEvent.CHANNEL_CLOSED, {
-            reason: REASONS.NOT_FOUND,
+            reason: REASONS.TIMEOUT,
           })
-          return
         }
-
-        // Reconnect
-        if (errorTime - this._startTime < 1000) {
+        if (!navigator.onLine) {
```
Comment (faiq-naufal) [inline packages/room/channel/channel.js:133]: @GagahGK , Why did you remove this `return` after timeout? From this logic, when the channel is closed because of timeout, the logic will continue to check the `!navigator.onLine` part. Is this intentional?

Comment (GagahGK) [reply packages/room/channel/channel.js:133]: ah I didn't realize the return was not written, thankyou for checking up on this changes

**Review by faiq-naufal — APPROVED** [review]
